test/stdlib/NewString.swift

```diff
@@ -81,23 +83,24 @@ func nonASCII() {
   // CHECK: --- UTF-16 basic round-tripping ---
   print("--- UTF-16 basic round-tripping ---")
 
-  // check that no extraneous objects are created
-  // CHECK-NEXT: __NSCFString@[[utf16address:[x0-9a-f]+]] = "🏂☃❅❆❄︎⛄️❄️"
+  // check that no extraneous objects are created when going from
+  // String to NSString and back (but native NSString to String can copy)
+  // CHECK-NEXT: __NSCFString@[[whatever:[x0-9a-f]+]] = "🏂☃❅❆❄︎⛄️❄️"
   print("  \(repr(nsUTF16))")
 
-  // CHECK-NEXT: String(Contiguous(owner: .cocoa@[[utf16address]], count: 11))
+  // CHECK-NEXT: String(Contiguous(owner: .native@[[utf16address:[x0-9a-f]+]][0...11], capacity = 16))
   var newNSUTF16 = nsUTF16 as String
   print("  \(repr(newNSUTF16))")
 
-  // CHECK-NEXT: __NSCFString@[[utf16address]] = "🏂☃❅❆❄︎⛄️❄️"
+  // CHECK-NEXT: _NSContiguousString@[[objcaddr:[x0-9a-f]+]] = "🏂☃❅❆❄︎⛄️❄️"
   var nsRoundTripUTF16 = newNSUTF16 as NSString
   print("  \(repr(nsRoundTripUTF16))")
 
   // CHECK: --- UTF-16 slicing ---
   print("--- UTF-16 slicing ---")
 
   // Slicing the String does not allocate
-  // CHECK-NEXT: String(Contiguous(owner: .cocoa@[[utf16address]], count: 6))
+  // CHECK-NEXT: String(Contiguous(owner: .native@[[utf16address]][-6...0], capacity = 16))
```

Comment:
Don't do equality checks on capacity, allocator behavior is different across platforms.